src/create-broadcast.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Creates a broadcast that can be listened to, i.e. simple event emitter
+ *
+ * @see https://github.com/ReactTraining/react-broadcast
+ */
+
+const createBroadcast = initialValue => {
+  let listeners = [];
+  let currentValue = initialValue;
+
+  return {
+    publish(value) {
+      currentValue = value;
+      listeners.forEach(listener => listener(currentValue));
+    },
+    subscribe(listener) {
+      listeners.push(listener);
+
+      // Publish to this subscriber once immediately.
+      listener(currentValue);
+
+      return () => {
+        listeners = listeners.filter(item => item !== listener);
+      };
+    },
+  };
+};
+
+module.exports = createBroadcast;
```

Comment:
@vesparny how does your `componentDidMount` implementation work with SSR? afaik it doesnt, because the `componentWillMount is the only lifecycle hook called on server rendering`. correct me if im wrong. We can switch from `componentWillMount` to `constructor`, so it works for isomorphic apps and yet aligned with react docs